File: transport-company-backend/src/main/java/com/example/transportcompanybackend/dto/DeliveryDto.java
package com.example.transportcompanybackend.dto;

import com.example.transportcompanybackend.entity.enums.DeliveryStatus;

import lombok.Data;

import java.sql.Timestamp;


@Data
public class DeliveryDto {
    private Long id;
    private WarehouseDto warehouseFrom;
    private WarehouseDto warehouseTo;
    private TransporterDto transporter;
    private String cargoName;
    private Double cargoAmount;
    private DeliveryStatus status;
    private Timestamp created;
    private Timestamp scheduled;
    private Timestamp actual;
}

File: transport-company-frontend/src/app/api/models/delivery-dto.ts
import { WarehouseDto } from "./warehouse-dto";
import { TransporterDto } from "./transporter-dto";
import { DeliveryStatus } from "./delivery-status";


export interface DeliveryDto {
  id: number;
  warehouse_from?: WarehouseDto;
  warehouse_to?: WarehouseDto;
  transporter?: TransporterDto;
  cargo_name: string;
  cargo_amount: number;
  status: DeliveryStatus;
  created?: string;
  scheduled?: string;
  actual?: string;
}
